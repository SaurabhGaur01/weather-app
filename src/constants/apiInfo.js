export const APP_ID = "75f972b80e26f14fe6c920aa6a85ad57";
export const TEMPERATURE_IN_FAHRENHEIT = "Imperial";
export const TEMPERATURE_IN_CELSIUS = "Metric";
export const COUNT = 40;
export const BASE_URL = "http://api.openweathermap.org/data/2.5/forecast";
export const DEFAULT_LOCATION = "Munich,de";
export const CELSIUS_UNIT = "°C";
export const FAHRENHEIT_UNIT = "°F";