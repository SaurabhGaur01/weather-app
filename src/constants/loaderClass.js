import { css } from "@emotion/core";

export const loaderClass = css`
  display: block;
  margin: 170px auto;
  border-color: red;
`;
export const loaderSize = 60;
export const loaderColor = '#123abc';