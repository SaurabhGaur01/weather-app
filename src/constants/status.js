export const API_CALL_STATUS_SUCCESS = "SUCCESS";
export const API_CALL_STATUS_LOADING = "LOADING";
export const API_CALL_STATUS_ERROR = "ERROR";