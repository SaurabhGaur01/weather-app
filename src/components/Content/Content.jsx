import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import RingLoader from "react-spinners/RingLoader";
import Temperature from './Temperature/Temperature';
import WeatherCard from './WeatherCard/WeatherCard';
import { contentClass } from '../../utils/getStyles';
import { API_CALL_STATUS_LOADING, API_CALL_STATUS_SUCCESS } from '../../constants/status';
import { loaderClass, loaderSize, loaderColor } from '../../constants/loaderClass';

const Content = ({apiCallStatus}) => {
    const classes = contentClass();
    return (
        <div class="container">
            <div class="row">
                <div className="col-md-2" />
                <div class="col-md-8">
                    <Card className={`root-container ${classes.root}`}>
                        <CardContent>
                            <RingLoader
                                css={loaderClass}
                                size={loaderSize}
                                color={loaderColor}
                                loading={apiCallStatus === API_CALL_STATUS_LOADING ? true: false}
                            />
                            <Temperature />
                            {
                                apiCallStatus === API_CALL_STATUS_SUCCESS && 
                                <WeatherCard />    
                            }
                        </CardContent>
                    </Card>
                </div>
                <div className="col-md-2" />
            </div>
        </div>            
    )
}

Content.propTypes = {
    apiCallStatus: PropTypes.string.isRequired,
};

const mapStateToProps = state => ({
    apiCallStatus: state.apiCallStatus,
});

export { Content as TestableContent };

export default connect(mapStateToProps)(Content);