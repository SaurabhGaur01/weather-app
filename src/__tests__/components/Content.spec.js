import React from 'react';
import { shallow } from 'enzyme';
import { TestableContent } from '../../components/Content/Content';
import { API_CALL_STATUS_LOADING, API_CALL_STATUS_SUCCESS } from '../../constants/status';

describe('Content Component', () => {
    let props;

    beforeEach(() =>{
        props = {
            apiCallStatus: API_CALL_STATUS_LOADING,
            contentClass: jest.fn(),
        }
    });

    it('should render correctly', () => {
        const renderedModule = shallow(<TestableContent {...props} />);
        expect(renderedModule).toMatchSnapshot();
    });

    it('should render correctly when apiCallStatus is Success', () => {
        props.apiCallStatus = API_CALL_STATUS_SUCCESS;
        const renderedModule = shallow(<TestableContent {...props} />);
        expect(renderedModule).toMatchSnapshot();
    });
})