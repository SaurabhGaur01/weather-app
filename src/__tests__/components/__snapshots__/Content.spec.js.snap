// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content Component should render correctly 1`] = `
<div
  class="container"
>
  <div
    class="row"
  >
    <div
      className="col-md-2"
    />
    <div
      class="col-md-8"
    >
      <[object Object]
        className="root-container makeStyles-root-1"
      >
        <[object Object]>
          <Loader
            color="#123abc"
            css={
              Object {
                "map": undefined,
                "name": "agvke8",
                "next": undefined,
                "styles": "
  display: block;
  margin: 170px auto;
  border-color: red;
",
                "toString": [Function],
              }
            }
            loading={true}
            size={60}
          />
          <[object Object] />
        </[object Object]>
      </[object Object]>
    </div>
    <div
      className="col-md-2"
    />
  </div>
</div>
`;

exports[`Content Component should render correctly when apiCallStatus is Success 1`] = `
<div
  class="container"
>
  <div
    class="row"
  >
    <div
      className="col-md-2"
    />
    <div
      class="col-md-8"
    >
      <[object Object]
        className="root-container makeStyles-root-1"
      >
        <[object Object]>
          <Loader
            color="#123abc"
            css={
              Object {
                "map": undefined,
                "name": "agvke8",
                "next": undefined,
                "styles": "
  display: block;
  margin: 170px auto;
  border-color: red;
",
                "toString": [Function],
              }
            }
            loading={false}
            size={60}
          />
          <[object Object] />
          <[object Object] />
        </[object Object]>
      </[object Object]>
    </div>
    <div
      className="col-md-2"
    />
  </div>
</div>
`;
